AWSCore/Authentication/AWSCredentialsProvider.m

```diff
@@ -636,10 +647,11 @@ - (void)setUpWithRegionType:(AWSRegionType)regionType
             // 1. The cached logins are different from the one the identity provider provided.
             // 2. The cached credentials is nil.
             // 3. The credentials expire within 10 minutes.
+            credentials = self.internalCredentials.copy;
```

Comment:
What is the reason you are assigning it back? Can you give a brief description?

Reply:
I am just holding on to the value of `self.internalCredentials` inside a local variable so that we will have a valid value  even if the class variable has changed.